pkg/generate/app/app.go

```diff
@@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
 		r.Name = name
 	}
 
-	selector := map[string]string{
-		"deploymentconfig": r.Name,
+	var selector map[string]string
+	if len(r.Labels) > 0 {
```

Comment:
yup you're right.... we just need the additional labels instead of replacing them